great_expectations/cli/suite.py

```diff
@@ -116,15 +182,52 @@ def suite_new(ctx, expectation_suite, interactive, profile, batch_request, no_ju
         )
         sys.exit(1)
 
-    _suite_new_workflow(
-        context=context,
-        expectation_suite_name=expectation_suite,
-        interactive=interactive,
-        profile=profile,
-        no_jupyter=no_jupyter,
-        usage_event=usage_event_end,
-        batch_request=batch_request,
-    )
+    # TODO: AJB 20210420 add better warning if user passes --non-interactive but either --profile or --batch-request (for this and suite_edit...)
+
+    # If user has provided a flag determining their configuration, skip prompt.
+    if (interactive is not None) or (profile is True) or (batch_request is not None):
+        # Assume batch needed if user passes --profile
+        if profile is True and (interactive is False or interactive is None):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --profile flag</green>"
+            )
+            interactive = True
+        # Assume batch needed if user passes --batch-request
+        elif (batch_request is not None) and (
+            interactive is False or interactive is None
+        ):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --batch-request flag</green>"
+            )
+            interactive = True
+    else:
+        suite_create_method = click.prompt(
+            """\
+How would you like to create your Expectation Suite?
+    1. Manually, without interacting with a sample batch of data (default)
+    2. Interactively, with a sample batch of data
+    3. Automatically, using a profiler
+""",
+            type=click.Choice(["1", "2", "3"]),
```

Comment:
@anthonyburdi I would switch the order: 1 should be Interactively (with a batch); 2 should be Automatically (using a profiler); and only 3 -- would be Manually (no data).  The rationale is that we want the users to taste the joys of interacting with data, as much as possible.  Would you agree, or was there a reason for making "no data" the default?  Thanks.

Reply:
I agree, but with this whole design I am trying to hit the middle ground between keeping with the intent of the spec (default no batch no prompt) and adding the prompt so that at least it is clear there are other options. So if a user does not specify interactive mode specifically, they still have a chance to enter it after the prompt.